cli/injector/main.go

```diff
@@ -329,44 +381,54 @@ func injectAndWait(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	// wait for an exit signal, this is a blocking call
-	sig := <-signals
+	if !errOnInject && pulseActiveDuration > 0 && pulseDormantDuration > 0 {
+		var action func(string, bool) bool
 
-	log.Infow("an exit signal has been received", "signal", sig.String())
-}
+		isInjected := true
+		sleepDuration := pulseActiveDuration
 
-// cleanAndExit cleans the disruption with the configured injector and exits nicely
-func cleanAndExit(cmd *cobra.Command, args []string) {
-	errs := []error{}
+		for { // This loop will wait, clean, wait, inject until a signal is received
+			select { // Quit on signal reception or sleep and injects / cleans the disruptions
+			case sig := <-signals:
+				log.Infow("an exit signal has been received", "signal", sig.String())
 
-	for _, inj := range injectors {
-		// start cleanup which is retried up to 3 times using an exponential backoff algorithm
-		if err := backoff.RetryNotify(inj.Clean, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3), retryNotifyHandler); err != nil {
-			handleMetricError(ms.MetricCleaned(false, cmd.Name(), nil))
-
-			errs = append(errs, err)
+				return
+			case <-time.After(sleepDuration):
+				if !isInjected {
+					action = inject
+					sleepDuration = pulseDormantDuration
+				} else {
+					action = clean
+					sleepDuration = pulseActiveDuration
+				}
+
+				if errOnClean := action(cmd.Name(), false); errOnClean {
```

Comment:
I'd like to make a small complaint. `errOnClean` is too misleading to use here, as we are alternating between Inject and Clean